### PR TITLE
Feature: conditional execution

### PR DESCRIPTION
**Pull request by artemii-yanushevskyi**

### CODE REVIEW

File: content/getting-started/yaml.md [hunk 1]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+        - push
+    when:
+      changeset:
+        includes:
+          - '.'
+        excludes:
+          - '**/*.md'
+```
+
+In this case, build would be skipped if there were no changes to the repository except Markdown files `.md`.
```
Comment (craastad) [inline content/getting-started/yaml.md:332]: Change to: `build would be skipped if there were changes only to the Markdown files `.md`.`

The current wording is confusing, `except` is hard to follow.

File: content/getting-started/yaml.md [hunk 2]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+
+Note that `codemagic.yaml` is always included in change set by default.
+
+Both keys `includes` and `excludes` in `changeset` are *optional*. If `includes` key is not specified, its value would defalut to `'.'`. `excludes` value defaults to an empty array.
```
Comment (craastad) [inline content/getting-started/yaml.md:336]: empty array is not a good docs terminology.
excludes defaults to nothing? or no exclusions?

File: content/getting-started/yaml.md [hunk 3]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+
+If you use a monorepo, each workflow could be responsible to building a part of your application, use conditional workflow execution. Specify the path to the application in change set as in the example below
```
Comment (craastad) [inline content/getting-started/yaml.md:338]: This is brilliant 😻

File: content/getting-started/yaml.md [hunk 4]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+    triggering:
+      events:
+        - push
+    when:
+      changeset:
+        includes:
+          - 'android/'
+```
+
+As a result, all commits don't affect `android` folder will not be built.
```
Comment (craastad) [inline content/getting-started/yaml.md:353]: As a result, all commits with changes outside of the `android` folder will not trigger a build.

File: content/getting-started/yaml.md [hunk 5]
```diff
@@ -307,3 +302,69 @@ scripts:
     echo 'This is a Post-publish script'
     echo 'This script is multiline'
 ```
+
+## Conditional execution
```
Comment (helinanever) [inline content/getting-started/yaml.md:306]: Let's add an intro:
You can skip building particular commits or watch for changes in specific files to trigger builds. It's possible to define build conditions per workflow or specific build steps.

Comment (craastad) [inline content/getting-started/yaml.md:306]: Maybe `Conditional build triggers`

File: content/getting-started/yaml.md [hunk 6]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+        - push
+    when:
+      changeset:
+        includes:
+          - 'android/'
+```
+
+As a result, all commits don't affect `android` folder will not be built.
+
+You may also want to skip some specific steps when building your application. Use the same approach with scripts
```
Comment (craastad) [inline content/getting-started/yaml.md:355]: We could break this into a section

`## Conditional build step execution`

File: content/getting-started/yaml.md [hunk 7]
```diff
@@ -307,3 +302,69 @@ scripts:
...
 ```
+
+## Conditional execution
+### Skip building a commit
+
+If you do not wish Codemagic to build a particular commit, include `[skip ci]` or `[ci skip]` in your commit message.
+
+### Watch for changes in files
+
+You can avoid unnecessary builds when functional components of your repository were not modified. Use conditional workflow execution to skip building workflow if watched files were not updated since the last successful build.
```
Comment (helinanever) [inline content/getting-started/yaml.md:313]: Use conditional workflow triggering to skip building the workflow if the watched files....

File: content/getting-started/yaml.md [hunk 8]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+
+You should specify files to watch in `changeset` by using `includes` and `excludes` keys.
```
Comment (helinanever) [inline content/getting-started/yaml.md:315]: ...specify the files...
...using the `includes` and `excludes` keys

File: content/getting-started/yaml.md [hunk 9]
```diff
@@ -307,3 +302,69 @@ scripts:
...
+      changeset:
+        includes:
+          - '.'
+        excludes:
+          - '**/*.md'
+```
+
+In this case, build would be skipped if there were no changes to the repository except Markdown files `.md`.
+
+Note that `codemagic.yaml` is always included in change set by default.
```
Comment (helinanever) [inline content/getting-started/yaml.md:334]: ...in the change set

File: content/getting-started/yaml.md (hunk 2 above)
Comment (helinanever) [inline content/getting-started/yaml.md:336]: If the `includes` key is not specified,...
The `excludes` key defaults to...

File: content/getting-started/yaml.md (hunk 3 above)
Comment (helinanever) [inline content/getting-started/yaml.md:338]: responsible for..
start a new sentence from here: Use conditional workflow triggering and specify the path to the application in the changeset as in the example below.

**Comment by craastad** [comment]
Let me know when these docs are finalized and I'll review them again. Good stuff here!

**Comment by artemii-yanushevskyi** [comment]
Addressed the comments, feel free to take another look

File: content/getting-started/yaml.md [hunk 10]
```diff
@@ -307,3 +302,73 @@ scripts:
...
+      changeset:
+        includes:
+          - '.'
+        excludes:
+          - '**/*.md'
+```
+
+In this case, build would be skipped if there were changes only to Markdown files `.md`.
+
+Note that `codemagic.yaml` is always included in the change set by default.
```
Comment (helinanever) [inline content/getting-started/yaml.md:336]: changeset

File: content/getting-started/yaml.md [hunk 11]
```diff
@@ -307,3 +302,73 @@ scripts:
...
+    triggering:
+      events:
+        - push
+    when:
+      changeset:
+        includes:
+          - 'android/'
+```
+
+As a result, all commits with changes outside of the android folder will not trigger a build.
```
Comment (helinanever) [inline content/getting-started/yaml.md:355]: remove "all"

**Review by helinanever — APPROVED** [review]
LGTM

**Review by craastad — APPROVED** [review]
